buildenv/jenkins/common/build.groovy

```diff
@@ -325,6 +327,25 @@ def archive_sdk() {
 }
 
 def archive_diagnostics() {
+    if (SPEC.contains('zos')) {
+   	 def logContent = currentBuild.rawBuild.getLog()
+    	 def match = logContent =~ /IEATDUMP success for DSN=.*/
+         def size = match.size()
+         def index = 0
+    def moveFile = {
+      def beginIndex = match[index].indexOf('.') + 1
+      def endIndex = match[index].length() - 1
+      def filename = match[index].substring(beginIndex,endIndex)
+      index = index + 1
+      try {
+         sh "mv '//${filename}' core.${filename}.dmp"
+      } catch (e) {
+         println "Dataset is unavailable"
+      }
+    }
+    1.upto(size, moveFile)
```

Comment:
Does this still work if there are no matches?

Reply:
If there are matches, it will show files not found.

Reply:
I don't follow.

Reply:
> I don't follow.

Sorry, it will show files not found if there are not matches